cli/js/web/workers.ts

```diff
@@ -28,6 +29,75 @@ function decodeMessage(dataIntArray: Uint8Array): any {
   return JSON.parse(dataJson);
 }
 
+export interface PostMessageOptions {
+  transfer: object[];
+}
+export interface MessagePort extends EventTarget {
+  postMessage(message: any, transfer: object[]): void;
+  postMessage(message: any, options: PostMessageOptions): void;
+}
+export type MessageEventSource = MessagePort | Worker; // | WindowProxy
+export interface MessageEventInit extends domTypes.EventInit {
+  data?: any;
+  origin?: string;
+  lastEventId?: string;
+  source?: MessageEventSource | null;
+  ports?: MessagePort[];
+}
+export class MessageEvent extends Event {
+  readonly data: any;
+  readonly origin: string;
+  readonly lastEventId: string;
+  readonly source: MessageEventSource | null;
+  //readonly ports: MessagePort[];
+
+  constructor(type: string, eventInitDict?: MessageEventInit) {
+    super(type, {
+      bubbles: eventInitDict.bubbles,
+      cancelable: eventInitDict.cancelable,
+      composed: eventInitDict.composed
+    });
+
+    if (eventInitDict.ports) {
+      notImplemented();
+    }
+
+    this.data = eventInitDict.data;
+    this.origin = eventInitDict.origin;
+    this.lastEventId = eventInitDict.lastEventId;
+    this.source = eventInitDict.source;
+  }
+}
```

Comment:
You're dereferencing `eventInitDict` which could be undefined. Use optional chaining, replace all the `.`s with `?.`s.

Reply:
Mind the types of the fields. Either they need to be nullable now or you need to specify a default like shown above.

Reply:
whoops